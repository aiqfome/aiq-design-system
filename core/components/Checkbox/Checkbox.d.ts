import React, { InputHTMLAttributes } from 'react';
import { DefaultTheme } from 'styled-components';
export interface Props extends DefaultTheme, InputHTMLAttributes<HTMLInputElement> {
    label?: string;
    labelColor?: string;
    style?: any;
}
export declare const Checkbox: React.ForwardRefExoticComponent<Props & React.RefAttributes<HTMLInputElement>>;
//# sourceMappingURL=Checkbox.d.ts.map