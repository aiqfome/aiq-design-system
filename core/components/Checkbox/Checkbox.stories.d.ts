import React from 'react';
declare const _default: {
    component: React.ForwardRefExoticComponent<import("./Checkbox").Props & React.RefAttributes<HTMLInputElement>>;
    title: string;
    decorators: ((...args: any) => any)[];
};
export default _default;
export declare const Basic: React.FC;
export declare const Checked: React.FC;
export declare const Disabled: React.FC;
export declare const DisabledChecked: React.FC;
//# sourceMappingURL=Checkbox.stories.d.ts.map