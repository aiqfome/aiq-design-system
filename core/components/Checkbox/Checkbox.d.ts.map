{"version":3,"file":"Checkbox.d.ts","sourceRoot":"","sources":["../../../src/components/Checkbox/Checkbox.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAK,EAAE,EAAE,mBAAmB,EAAE,MAAM,OAAO,CAAA;AAGlD,OAAe,EAAE,YAAY,EAAE,MAAM,mBAAmB,CAAA;AAGxD,MAAM,WAAW,KACf,SAAQ,YAAY,EAClB,mBAAmB,CAAC,gBAAgB,CAAC;IACvC,KAAK,CAAC,EAAE,MAAM,CAAA;IACd,UAAU,CAAC,EAAE,MAAM,CAAA;IACnB,KAAK,CAAC,EAAE,GAAG,CAAA;CACZ;AAiDD,eAAO,MAAM,QAAQ,gFAwBpB,CAAA"}