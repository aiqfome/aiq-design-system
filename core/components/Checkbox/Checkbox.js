"use strict";
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Checkbox = void 0;
const react_1 = __importDefault(require("react"));
const prop_types_1 = __importDefault(require("prop-types"));
const styled_components_1 = __importDefault(require("styled-components"));
const styled_system_1 = require("styled-system");
const Label = styled_components_1.default.label `
  ${styled_system_1.color}

  display: flex;
  align-items: center;

  opacity: ${({ disabled }) => (disabled ? 0.5 : 1)};
`;
const HiddenCheckbox = styled_components_1.default.input `
  border: 0;
  display: none;
`;
const BoxCheckbox = styled_components_1.default.div `
  display: inline-block;
  margin-right: 12px;
  width: 16px;
  height: 16px;
  border-radius: 2px;
  border: 2px solid ${({ theme }) => theme.colors.primary};

  transition: all 0.2s;

  background: transparent;

  ${HiddenCheckbox}:checked + & {
    background: ${({ theme }) => theme.colors.primary};
  }

  ${HiddenCheckbox}:focus + & {
    box-shadow: 0 0 0 3px ${({ theme }) => theme.colors.primary};
  }
`;
const Icon = styled_components_1.default.svg `
  display: none;
  fill: none;
  stroke: white;
  stroke-width: 4px;
  transition: all 0.2s;

  ${HiddenCheckbox}:checked ~ div >  & {
    display: block;
  }
`;
exports.Checkbox = react_1.default.forwardRef((_a, ref) => {
    var { label, style, labelColor, disabled } = _a, props = __rest(_a, ["label", "style", "labelColor", "disabled"]);
    return (react_1.default.createElement(Label, { "data-testid": 'checkbox', color: labelColor, style: style, disabled: disabled },
        react_1.default.createElement(HiddenCheckbox, Object.assign({ disabled: disabled, type: 'checkbox', ref: ref }, props)),
        react_1.default.createElement(BoxCheckbox, null,
            react_1.default.createElement(Icon, { viewBox: '0 0 24 24' },
                react_1.default.createElement("polyline", { points: '20 6 9 17 4 12' }))),
        label));
});
exports.Checkbox.displayName = 'Checkbox';
exports.Checkbox.propTypes = {
    checked: prop_types_1.default.bool,
    label: prop_types_1.default.string,
    labelColor: prop_types_1.default.string,
    style: prop_types_1.default.any,
    disabled: prop_types_1.default.bool
};
