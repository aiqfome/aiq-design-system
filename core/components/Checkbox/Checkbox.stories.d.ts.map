{"version":3,"file":"Checkbox.stories.d.ts","sourceRoot":"","sources":["../../../src/components/Checkbox/Checkbox.stories.tsx"],"names":[],"mappings":"AAAA,OAAO,KAAmB,MAAM,OAAO,CAAA;;;;;;AAMvC,wBAIC;AAED,eAAO,MAAM,KAAK,EAAE,KAAK,CAAC,EAkBzB,CAAA;AAED,eAAO,MAAM,OAAO,EAAE,KAAK,CAAC,EAY3B,CAAA;AAED,eAAO,MAAM,QAAQ,EAAE,KAAK,CAAC,EAa5B,CAAA;AAED,eAAO,MAAM,eAAe,EAAE,KAAK,CAAC,EAanC,CAAA"}