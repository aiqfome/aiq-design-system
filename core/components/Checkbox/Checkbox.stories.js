"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.DisabledChecked = exports.Disabled = exports.Checked = exports.Basic = void 0;
const react_1 = __importStar(require("react"));
const addon_knobs_1 = require("@storybook/addon-knobs");
const Flex_1 = require("../Flex");
const Checkbox_1 = require("./Checkbox");
exports.default = {
    component: Checkbox_1.Checkbox,
    title: 'Checkbox',
    decorators: [addon_knobs_1.withKnobs]
};
exports.Basic = () => {
    const [checked, setChecked] = react_1.useState(false);
    function handleOnChangeChange(event) {
        setChecked(event.target.checked);
    }
    return (react_1.default.createElement(Flex_1.Flex, { variant: 'fullCentralized' },
        react_1.default.createElement(Checkbox_1.Checkbox, { onChange: handleOnChangeChange, checked: checked, disabled: addon_knobs_1.boolean('disabled', false), label: addon_knobs_1.text('text', 'Aiqfome'), labelColor: 'primary' })));
};
exports.Checked = () => {
    const [checked, setChecked] = react_1.useState(true);
    return (react_1.default.createElement(Flex_1.Flex, { variant: 'fullCentralized' },
        react_1.default.createElement(Checkbox_1.Checkbox, { onChange: event => setChecked(event.target.checked), checked: checked, label: addon_knobs_1.text('text', 'Aiqfome') })));
};
exports.Disabled = () => {
    const [checked, setChecked] = react_1.useState(false);
    return (react_1.default.createElement(Flex_1.Flex, { variant: 'fullCentralized' },
        react_1.default.createElement(Checkbox_1.Checkbox, { onChange: event => setChecked(event.target.checked), disabled: true, checked: checked, label: addon_knobs_1.text('text', 'Aiqfome') })));
};
exports.DisabledChecked = () => {
    const [checked, setChecked] = react_1.useState(true);
    return (react_1.default.createElement(Flex_1.Flex, { variant: 'fullCentralized' },
        react_1.default.createElement(Checkbox_1.Checkbox, { onChange: event => setChecked(event.target.checked), disabled: true, checked: checked, label: addon_knobs_1.text('text', 'Aiqfome') })));
};
