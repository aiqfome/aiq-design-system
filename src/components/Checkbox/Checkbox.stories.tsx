import React, { useState } from 'react'
import { withKnobs, boolean, text } from '@storybook/addon-knobs'

import { Flex } from '../Flex'
import { Checkbox } from './Checkbox'

export default {
  component: Checkbox,
  title: 'Checkbox',
  decorators: [withKnobs]
}

export const Basic: React.FC = () => {
  const [checked, setChecked] = useState(false)

  function handleOnChangeChange(event) {
    setChecked(event.target.checked)
  }

  return (
    <Flex variant='fullCentralized'>
      <Checkbox
        onChange={handleOnChangeChange}
        checked={checked}
        disabled={boolean('disabled', false)}
        label={text('text', 'Aiqfome')}
        labelColor='primary'
      />
    </Flex>
  )
}

export const Checked: React.FC = () => {
  const [checked, setChecked] = useState(true)

  return (
    <Flex variant='fullCentralized'>
      <Checkbox
        onChange={event => setChecked(event.target.checked)}
        checked={checked}
        label={text('text', 'Aiqfome')}
      />
    </Flex>
  )
}

export const Disabled: React.FC = () => {
  const [checked, setChecked] = useState(false)

  return (
    <Flex variant='fullCentralized'>
      <Checkbox
        onChange={event => setChecked(event.target.checked)}
        disabled={true}
        checked={checked}
        label={text('text', 'Aiqfome')}
      />
    </Flex>
  )
}

export const DisabledChecked: React.FC = () => {
  const [checked, setChecked] = useState(true)

  return (
    <Flex variant='fullCentralized'>
      <Checkbox
        onChange={event => setChecked(event.target.checked)}
        disabled={true}
        checked={checked}
        label={text('text', 'Aiqfome')}
      />
    </Flex>
  )
}
