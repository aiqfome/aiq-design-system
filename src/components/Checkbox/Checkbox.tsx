import React, { InputHTMLAttributes } from 'react'
import PropTypes from 'prop-types'

import styled, { DefaultTheme } from 'styled-components'
import { color } from 'styled-system'

export interface Props
  extends DefaultTheme,
    InputHTMLAttributes<HTMLInputElement> {
  label?: string
  labelColor?: string
  style?: any
}

const Label = styled.label<Props>`
  ${color}

  display: flex;
  align-items: center;

  opacity: ${({ disabled }) => (disabled ? 0.5 : 1)};
`

const HiddenCheckbox = styled.input<Props>`
  border: 0;
  display: none;
`

const BoxCheckbox = styled.div<Props>`
  display: inline-block;
  margin-right: 12px;
  width: 16px;
  height: 16px;
  border-radius: 2px;
  border: 2px solid ${({ theme }) => theme.colors.primary};

  transition: all 0.2s;

  background: transparent;

  ${HiddenCheckbox}:checked + & {
    background: ${({ theme }) => theme.colors.primary};
  }

  ${HiddenCheckbox}:focus + & {
    box-shadow: 0 0 0 3px ${({ theme }) => theme.colors.primary};
  }
`

const Icon = styled.svg<Props>`
  display: none;
  fill: none;
  stroke: white;
  stroke-width: 4px;
  transition: all 0.2s;

  ${HiddenCheckbox}:checked ~ div >  & {
    display: block;
  }
`

export const Checkbox = React.forwardRef<HTMLInputElement, Props>(
  ({ label, style, labelColor, disabled, ...props }, ref) => {
    return (
      <Label color={labelColor} style={style} disabled={disabled}>
        <HiddenCheckbox
          disabled={disabled}
          type='checkbox'
          ref={ref}
          {...props}
        />
        <BoxCheckbox>
          <Icon viewBox='0 0 24 24'>
            <polyline points='20 6 9 17 4 12' />
          </Icon>
        </BoxCheckbox>
        {label}
      </Label>
    )
  }
)

Checkbox.displayName = 'Checkbox'

Checkbox.propTypes = {
  checked: PropTypes.bool,
  label: PropTypes.string,
  labelColor: PropTypes.string,
  style: PropTypes.any,
  disabled: PropTypes.bool
}
