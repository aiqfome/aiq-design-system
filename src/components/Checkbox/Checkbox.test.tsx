import React from 'react'

import { Checkbox } from './Checkbox'

import { render } from '../utils/test/render'

describe('must match with the previous snapshot', () => {
  test('snapshot renders', () => {
    const component = render(<Checkbox />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
