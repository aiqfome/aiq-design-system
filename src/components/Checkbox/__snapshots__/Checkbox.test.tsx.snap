// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`must match with the previous snapshot snapshot renders 1`] = `
<label
  className="sc-AxjAm hkOyki"
>
  <input
    className="sc-AxirZ eRnVJk"
    type="checkbox"
  />
  <div
    className="sc-AxiKw cwLquZ"
  >
    <svg
      className="sc-AxhCb cfxuFI"
      viewBox="0 0 24 24"
    >
      <polyline
        points="20 6 9 17 4 12"
      />
    </svg>
  </div>
</label>
`;
